src/utils.ts

```diff
@@ -82,9 +88,10 @@ export function loadScript(
 
   script.addEventListener('load', scriptLoadListener);
 
-  window.dataLayer = window.dataLayer ?? [];
+  const dataLayerName: string = config.dataLayerName ?? 'dataLayer';
+  window[dataLayerName] = window[dataLayerName] ?? [];
 
-  window.dataLayer?.push({
+  window[dataLayerName]?.push({
```

Comment:
mh :thinking: not really related to your MR, but I think we could safely use

```suggestion
  window[dataLayerName]!.push({
```

here

Reply:
Sorted this out by getting or initializing dataLayer in reusable function